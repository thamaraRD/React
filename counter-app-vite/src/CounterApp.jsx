import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleCount = () => {
    setCounter(counter + 1);
  };
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={handleCount}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
};
