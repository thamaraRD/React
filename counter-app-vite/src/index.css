html, body {
  background-color: #212328;
  color: white;
  font-size: 1.3rem;
  padding: 50px;
  font-family: Arial, Helvetica, sans-serif;
}

button{
  font-size: 1.3rem;
  padding: 15px;
  margin-right: 5px;
}