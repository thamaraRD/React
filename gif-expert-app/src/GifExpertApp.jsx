
export const GifExpertApp = () => {
  return (
    <>
    <h1>gifExpertApp</h1>
    </>
  )
}
