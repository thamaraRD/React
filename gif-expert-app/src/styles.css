* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 60px;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(32, 63, 155);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}
.loader {
  width: 48px;
  height: 48px;
  border: 5px solid #FFF;
  border-radius: 50%;
  display: inline-block;
  box-sizing: border-box;
  position: relative;
  animation: pulse 1s linear infinite;
}
.loader:after {
  content: '';
  position: absolute;
  width: 48px;
  height: 48px;
  border: 5px solid #FFF;
  border-radius: 50%;
  display: inline-block;
  box-sizing: border-box;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  animation: scaleUp 1s linear infinite;
}

@keyframes scaleUp {
  0% { transform: translate(-50%, -50%) scale(0) }
  60% , 100% { transform: translate(-50%, -50%)  scale(1)}
}
@keyframes pulse {
  0% , 60% , 100%{ transform:  scale(1) }
  80% { transform:  scale(1.2)}
}