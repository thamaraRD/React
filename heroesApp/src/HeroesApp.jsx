import { AuthProvider } from "./auth/context/AuthProvider"
import { AppRoutes } from "./router/AppRoutes"


export const HeroesApp = () => {
  return (
    <AuthProvider>

    <AppRoutes />
    
    </AuthProvider>
  )
}
