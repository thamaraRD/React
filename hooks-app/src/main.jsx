import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
// import { HooksApp } from './HooksApp';
// import { CounterApp } from './01-useState/CounterApp';
// import { CounterWithCustomHooks } from './01-useState/CounterWithCustomHooks';
// import { SimpleForm } from './02-useEffect/SimpleForm';
// import { SimpleFormWithCustomHooks } from './02-useEffect/SimpleFormWithCustomHooks';
// import { MultipleCustoomHook } from './03-examples/MultipleCustoomHook';
// import { FocusScreen } from './04-useRef/FocusScreen';
// import { Layout } from './05-useLayoutEffect/Layout';
// import { Memorize } from './06-memos/Memorize';
// import { MemoHook } from './06-memos/MemoHook';
// import { CallbackHook } from './06-memos/CallbackHook';
// import { Padre } from './07-tarea-memo/Padre';
// import './08-useReducer/intro-reducer';
// import { TodoApp } from './08-useReducer/TodoApp';
import { MainApp } from './09-useContext/MainApp';


ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
    <MainApp />
  /* </React.StrictMode> */
)
