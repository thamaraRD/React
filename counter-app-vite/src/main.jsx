import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "../src/components/CounterApp";
import { FirstApp } from '../src/components/FirstApp';
import './index.css';

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <CounterApp value={10}/>
    {/* <FirstApp title="Hola soy Thamara" /> */}
  </React.StrictMode>
);
