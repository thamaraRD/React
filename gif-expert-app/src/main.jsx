import React from 'react';
import ReactDOM from "react-dom/client";
import { GifExpertApp } from "../src/components/GifExpertApp";
import './styles.css';

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>
);
