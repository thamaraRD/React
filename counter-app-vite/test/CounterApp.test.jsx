import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('test in the <CounterApp />', () => { 
  const value = 100;

  test('should hacer match con el snapshot', () => { 

    const { container } = render( <CounterApp value={value} /> );
    
    expect(container).toMatchSnapshot();

  });

  test('should mostrar el valor inicial de 100', () => { 

    render( <CounterApp value={100} /> );

    // expect((value)).toBeTruthy();
    expect(screen.getByText(100)).toBeTruthy();

    });

  test('should incrementar con el boton +1', () => { 

    render( <CounterApp value={value} /> );
    fireEvent.click( screen.getByText('+1') );
    
    expect( screen.getByText('101') ).toBeTruthy();

  });

  test('should decrementar con el boton -1', () => { 

    render( <CounterApp value={value} /> );
    fireEvent.click( screen.getByText('-1') );
    
    expect( screen.getByText('99') ).toBeTruthy();

  });

  test('should funcionar el boton de Reset', () => { 

    render( <CounterApp value={value} /> );
    fireEvent.click( screen.getByText('+1') );
    fireEvent.click( screen.getByText('+1') );
    // fireEvent.click( screen.getByText('reset') );
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
    
    expect( screen.getByText(value) ).toBeTruthy();

  });

});