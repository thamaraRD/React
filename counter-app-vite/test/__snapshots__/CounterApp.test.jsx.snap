// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in the <CounterApp /> should hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
